src/shared/extensionGlobals.ts

```diff
@@ -16,63 +16,95 @@ import { SchemaService } from './schemas'
 import { TelemetryService } from './telemetry/telemetryService'
 import { Window } from './vscode/window'
 
+type Clock = Pick<
+    typeof globalThis,
+    'setTimeout' | 'setImmediate' | 'setInterval' | 'clearTimeout' | 'clearImmediate' | 'clearInterval' | 'Date'
+>
+
+/**
+ * Copies all *enumerable* properties from the global object.
+ * Some properties have to be added manually depending on how they exist on the prototype.
+ */
+function copyClock(): Clock {
+    return { ...globalThis, Date, Promise } as Clock
+}
+
+const globals = {} as ToolkitGlobals
+
+export function checkDidReload(context: ExtensionContext): boolean {
+    return !!context.globalState.get<string>('ACTIVATION_LAUNCH_PATH_KEY')
+}
+
+export function initialize(context: ExtensionContext, window: Window): ToolkitGlobals {
+    // TODO: we should throw here if already assigned. A few tests actually depend on the combined state
+    // of the extension activating plus test setup, so for now we have to do it like this :(
+
+    Object.assign(globals, {
+        context,
+        window,
+        clock: copyClock(),
+        didReload: checkDidReload(context),
+        iconPaths: { dark: {}, light: {} } as ToolkitGlobals['iconPaths'],
+        manifestPaths: {} as ToolkitGlobals['manifestPaths'],
+        visualizationResourcePaths: {} as ToolkitGlobals['visualizationResourcePaths'],
+    })
+
+    return globals
+}
+
+export default globals
+
 /**
  * Namespace for common variables used globally in the extension.
  * All variables here must be initialized in the activate() method of extension.ts
  */
-export namespace ext {
```

Comment:
I removed the namespace since namespaces are mostly considered 'legacy', at least for Node applications. Best just to use ES6 module syntax directly.

Reply:
Thanks. This has been on our TODO list for awhile :)